search/views.py

```diff
@@ -44,20 +43,51 @@
 
 
 def search(request):
-    query_params, tvars = search_prepare_parameters(request)
-    
-    # get sound ids of the requested cluster
+    query_params, advanced_search_params_dict, extra_vars = search_prepare_parameters(request)
+
+    # get the url query params for later sending it to the clustering engine
+    url_query_params_string = request.META['QUERY_STRING']
+
+    # get sound ids of the requested cluster when cluster filter facet
     cluster_id = request.GET.get('cluster_id', "")
-    in_ids = get_ids_in_cluster(request, cluster_id)
-    query_params.update({'in_ids': in_ids})  # for added cluster facet filter
+    in_ids = _get_ids_in_cluster(request, cluster_id)
```

Comment:
If I understand correctly, `in_ids` should only be loaded when there is a clustering faced active. I see in `_get_ids_in_cluster` you check if `cluster_id` is "" and if that is the case you return empty list. I think it would be better to move the logic here. You could do:
```
cluster_id = request.GET.get('cluster_id')  # <- this will be None if cluster_id is not set
in_ids = []
if cluster_id is not None:
    in_ids = _get_ids_in_cluster(request, cluster_id)
```

Reply:
Also, where is the `in_ids` used? Inside `perform_solr_query`? if that is the case, add a comment as well `this is used in perform_solr_query to filter the sound ids included in the search